server/task_executor_schema_update_ghost_sync.go

```diff
@@ -241,23 +246,23 @@ func (exec *SchemaUpdateGhostSyncTaskExecutor) runGhostMigration(_ context.Conte
 	}(ctx)
 
 	go func() {
-		if err := migrator.Migrate(); err != nil {
+		err := migrator.Migrate()
+		if err != nil {
 			log.Error("failed to run gh-ost migration", zap.Error(err))
-			// `migrator.Migrate` can return an error.
-			// 1. If the error is returned before closing `syncDone`,`runGhostMigration` will receive from `syncError`.
-			// 2. If it is returned after `syncDone` is closed, `runGhostMigration` would have received from `syncDone` and returned,
-			// and there's no consumer for `syncError`, so we must send to `syncError` without blocking.
-			select {
-			case syncError <- fmt.Errorf("failed to run gh-ost migration, error: %w", err):
-			default:
-			}
+			migrationError <- err
+			return
 		}
+		migrationError <- nil
+		// we send to migrationError channel anyway because
+		// before syncDone, the gh-ost sync task will receive it.
+		// after syncDone, the gh-ost cutover task will receive it.
```

Comment:
```suggestion
		// we send to migrationError channel anyway because:
		// 1. before syncDone, the gh-ost sync task will receive it.
		// 2. after syncDone, the gh-ost cutover task will receive it.
```